tests/main/hidden-snap-dir/task.yaml

```diff
@@ -184,3 +184,23 @@ execute: |
     check_env "--with-exposed-home" "x3"
     check_dirs "--with-exposed-home" "x3"
     check_data "--with-exposed-home" "x3" "$data"
+
+    echo "Check fresh install of core22 based snap"
+    snap remove --purge "$NAME"
+    rm -rf "$HOME"/Snap "$HOME"/.snap/data "$HOME/snap/$NAME"
+
+    snap install --dangerous "$NAME"_1.0_all.snap
+
+    check_env "--with-exposed-home" "x1"
+    check_dirs "--with-exposed-home" "x1"
+
+    "$NAME".cmd sh -c 'true'
+
+    echo "Check remove of freshly installed core22 based snap"
+    snap remove --purge "$NAME"
```

Comment:
should we write some data in the relevant dirs before this to check which data is kept/not kept? all the dirs should be freshly recreated/emptied on the install, right?

Reply:
Running the snap creates the dirs and symlink under ~/.snap/data/<name> so we can just add that check after 'snap remove'. Although that's testing 'snap remove' so if we eventually change it so it removes the directory then this will have to change as well

Reply:
I'm confused, I thought remove didn't remove those dirs?

Reply:
Remove deletes the revision and 'common' subdirs but leaves the symlink and the "outer" directory (e.g., ~/.snap/data/snap_name/)